arch/xtensa/include/xtensa-asm2-context.h

```diff
@@ -64,85 +70,168 @@
  * original/interrupted stack pointer.
  */
 
-#define BASE_SAVE_AREA_SIZE_COMMON	44
-#define BASE_SAVE_AREA_SIZE_EXCCAUSE	4
-
-#if XCHAL_HAVE_LOOPS
-#define BASE_SAVE_AREA_SIZE_LOOPS	12
-#else
-#define BASE_SAVE_AREA_SIZE_LOOPS	0
-#endif
+#ifndef __ASSEMBLER__
 
-#if XCHAL_HAVE_S32C1I
-#define BASE_SAVE_AREA_SIZE_SCOMPARE	4
-#else
-#define BASE_SAVE_AREA_SIZE_SCOMPARE	0
-#endif
+#include <stdint.h>
+#include <zephyr/toolchain.h>
 
-#if XCHAL_HAVE_THREADPTR && defined(CONFIG_THREAD_LOCAL_STORAGE)
-#define BASE_SAVE_AREA_SIZE_THREADPTR	4
-#else
-#define BASE_SAVE_AREA_SIZE_THREADPTR	0
+/**
+ * Base Save Area (BSA) during interrupt.
+ *
+ * This saves the registers during interrupt entrance
+ * so they can be restored later.
+ *
+ * Note that only A0-A3 are saved here. High registers
+ * are saved after the BSA.
+ */
+struct xtensa_irq_base_save_area {
+#if XCHAL_HAVE_FP && defined(CONFIG_CPU_HAS_FPU) && defined(CONFIG_FPU_SHARING)
+	uintptr_t fpu0;
+	uintptr_t fpu1;
+	uintptr_t fpu2;
```

Comment:
Do you mind renaming fpu0 and fpu1 to FCR and FSR to match the register name?

Reply:
Renamed.